### PR TITLE
dock and controller logic for access api

### PR DESCRIPTION
**Pull request by PravinRanjan10**
**What this PR does / why we need it**:
Add the dock and controller logic for access fileshare

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/controllers/fileshare.go [hunk 1]
```diff
@@ -70,6 +71,27 @@ func (f *FileSharePortal) CreateFileShareAcl() {
...
+		return
+	}
+
+	defer f.CtrClient.Close()
+
+	opt := &pb.CreateFileShareAclOpts{
+		Id:               result.Id,
+		Description:      result.Description,
+    Type:             result.Type,
+		AccessCapability: result.AccessCapability,
```
Comment (wisererik) [inline pkg/api/controllers/fileshare.go]: format this file

File: pkg/controller/selector/selector.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ func (s *selector) SelectSupportedPoolForFileShare(in *model.FileShareSpec) (*mo
 			filterRequest["id"] = in.PoolId
 		}
 
-		filterRequest["storageType"] = prf.StorageType
+		//filterRequest["storageType"] = prf.StorageType
```
Comment (wisererik) [inline pkg/controller/selector/selector.go]: why comment code here?

File: pkg/controller/selector/selector.go [hunk 3]
```diff
@@ -246,10 +246,10 @@ func (s *selector) SelectSupportedPoolForFileShare(in *model.FileShareSpec) (*mo
...
 				}
-				if !ds.IsEmptyStorageAccessCapability() {
-					filterRequest["extras.dataStorage.storageAccessCapability"] =
-						ds.StorageAccessCapability
-				}
+				//if !ds.IsEmptyStorageAccessCapability() {
+				//	filterRequest["extras.dataStorage.storageAccessCapability"] =
+				//		ds.StorageAccessCapability
+				//}
 				if ds.MaxFileNameLengthBytes != 0 {
```
Comment (wisererik) [inline pkg/controller/selector/selector.go]: same here

File: pkg/model/proto/model.proto [hunk 4]
```diff
@@ -716,6 +722,30 @@ message DetachVolumeOpts {
...
+    // The fileshareId
+    string fileshareId = 2;
+    string description = 3;
+    // The locality that file share belongs to, required.
+    // The type of access. Ex: IP based.
+    string Type = 4
+    // The accessCapability for fileshare.
+    string accessCapability = 5
+    // accessTo of the fileshare.
+    accessTo = 6
```
Comment (wisererik) [inline pkg/model/proto/model.proto]: what's the type of AccessTo?

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 5]
```diff
@@ -80,6 +80,11 @@ func (d *Driver) Setup() error {
 
 func (*Driver) Unset() error { return nil }
 
+func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (fshare *model.FileShareAclSpec, err error) {
+	// need to implement
+return fshare, nil
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: The style is not correct.

File: pkg/api/controllers/fileshare.go [hunk 6]
```diff
@@ -70,6 +71,27 @@ func (f *FileSharePortal) CreateFileShareAcl() {
...
+
+	defer f.CtrClient.Close()
+
+	opt := &pb.CreateFileShareAclOpts{
+		Id:               result.Id,
+		Description:      result.Description,
+                Type:             result.Type,
+		AccessCapability: result.AccessCapability,
+		AccessTo:         result.AccessTo,
+		Context:   ctx.ToJson(),
```
Comment (leonwanghui) [inline pkg/api/controllers/fileshare.go]: The code style is not right.

File: pkg/controller/controller.go [hunk 7]
```diff
@@ -957,6 +957,47 @@ func (c *Controller) DeleteFileShare(contx context.Context, opt *pb.DeleteFileSh
 	return pb.GenericResponseResult(nil), err
 }
 
+// CreateFileShare implements pb.ControllerServer.CreateFileShare
+func (c *Controller) CreateFileShareAcl(contx context.Context, opt *pb.CreateFileShareAclOpts) (*pb.GenericResponse, error) {
+	var err error
+	log.Info("controller server receive create file share request, vr =", opt)
+  ctx := osdsCtx.NewContextFromJson(opt.GetContext())
```
Comment (leonwanghui) [inline pkg/controller/controller.go]: Same here.

File: pkg/controller/fileshare/filesharecontroller.go [hunk 8]
```diff
@@ -57,6 +59,35 @@ type controller struct {
 	DockInfo *model.DockSpec
 }
 
+func (c *controller) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error) {
+	if err := c.Client.Connect(c.DockInfo.Endpoint); err != nil {
+		log.Error("when connecting dock client:", err)
+		return nil, err
+	}
+
+	response, err := c.Client.CreateFileShareAcl(context.Background(), (opt))
```
Comment (leonwanghui) [inline pkg/controller/fileshare/filesharecontroller.go]: I don't understand why there is (opt) here

File: pkg/model/proto/model.proto [hunk 9]
```diff
@@ -716,6 +722,30 @@ message DetachVolumeOpts {
...
+// CreateFileShareAclOpts is a structure which indicates all required properties for creating a file share.
+message CreateFileShareAclOpts {
+    // The uuid of the file share, optional when creating.
+    string id = 1;
+    // The fileshareId
+    string fileshareId = 2;
+    string description = 3;
+    // The locality that file share belongs to, required.
+    // The type of access. Ex: IP based.
+    string Type = 4
```
Comment (leonwanghui) [inline pkg/model/proto/model.proto]: s/Type/type

File: pkg/model/proto/model.proto [hunk 10]
```diff
@@ -716,6 +722,30 @@ message DetachVolumeOpts {
...
+    // accessTo of the fileshare.
+    repeated string accessTo = 6
+    // The Context
+    string context = 6;
+    // The Serialized profile
+    string profile = 7;
+    // The driverName
+    string driverName = 8;
+    // The Context
+    string context = 9;
```
Comment (leonwanghui) [inline pkg/model/proto/model.proto]: context is duplicated here.

**Pull request closed without merging**